src/ir_Mitsubishi.cpp

```diff
@@ -1186,3 +1198,449 @@ String IRMitsubishi136::toString(void) {
+    case stdAc::swingh_t::kWide: return kMitsubishi112SwingHWide;
+    case stdAc::swingh_t::kAuto: return kMitsubishi112SwingHAuto;
+    default: return kMitsubishi112SwingHAuto;
+  }
+}
+
+// Convert a native mode to it's common equivalent.
+stdAc::opmode_t IRMitsubishi112::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kMitsubishi112Cool: return stdAc::opmode_t::kCool;
+    case kMitsubishi112Heat: return stdAc::opmode_t::kHeat;
+    case kMitsubishi112Dry: return stdAc::opmode_t::kDry;
+    default: return stdAc::opmode_t::kAuto;
+  }
+}
+
+// Convert a native fan speed to it's common equivalent.
+stdAc::fanspeed_t IRMitsubishi112::toCommonFanSpeed(const uint8_t speed) {
+  switch (speed) {
+    case kMitsubishi112FanMax: return stdAc::fanspeed_t::kMax;
+    case kMitsubishi112FanMed: return stdAc::fanspeed_t::kMedium;
+    case kMitsubishi112FanLow: return stdAc::fanspeed_t::kLow;
+    case kMitsubishi112FanMin: return stdAc::fanspeed_t::kMin;
+    default: return stdAc::fanspeed_t::kMedium;
+  }
+}
+
+// Convert a native vertical swing to it's common equivalent.
+stdAc::swingv_t IRMitsubishi112::toCommonSwingV(const uint8_t pos) {
+  switch (pos) {
+    case kMitsubishi112SwingVHighest: return stdAc::swingv_t::kHighest;
+    case kMitsubishi112SwingVHigh: return stdAc::swingv_t::kHigh;
+    case kMitsubishi112SwingVMiddle: return stdAc::swingv_t::kMiddle;
+    case kMitsubishi112SwingVLow: return stdAc::swingv_t::kLow;
+    case kMitsubishi112SwingVLowest: return stdAc::swingv_t::kLowest;
+    default: return stdAc::swingv_t::kAuto;
+  }
+}
+
+// Convert a native vertical swing to it's common equivalent.
+stdAc::swingh_t IRMitsubishi112::toCommonSwingH(const uint8_t pos) {
+  switch (pos) {
+    case kMitsubishi112SwingHLeftMax: return stdAc::swingh_t::kLeftMax;
+    case kMitsubishi112SwingHLeftInner: return stdAc::swingh_t::kLeft;
+    case kMitsubishi112SwingHMiddle: return stdAc::swingh_t::kMiddle;
+    case kMitsubishi112SwingHRightInner: return stdAc::swingh_t::kRight;
+    case kMitsubishi112SwingHRightMax: return stdAc::swingh_t::kRightMax;
+    case kMitsubishi112SwingHWide: return stdAc::swingh_t::kWide;
+    case kMitsubishi112SwingHAuto: return stdAc::swingh_t::kAuto;
+    default: return stdAc::swingh_t::kAuto;
+  }
+}
+
+
+// Convert the A/C state to it's common equivalent.
+stdAc::state_t IRMitsubishi112::toCommon(void) {
+  stdAc::state_t result;
+  result.protocol = decode_type_t::MITSUBISHI112;
+  result.model = -1;  // No models used.
+  result.power = this->getPower();
+  result.mode = this->toCommonMode(this->getMode());
+  result.celsius = true;
+  result.degrees = this->getTemp();
+  result.fanspeed = this->toCommonFanSpeed(this->getFan());
+  result.swingv = this->toCommonSwingV(this->getSwingV());
+  result.quiet = this->getQuiet();
+  result.swingh = this->toCommonSwingH(this->getSwingH());;
+  result.econo = false;  // Need to figure this part from stdAc
```

Comment:
Do you need help with this?
`result.econo` is just a boolean flag. It's typically `true` for if you want the A/Cs economy mode on, and `false` for off.
If you don't have it, safest is to assume "Off". i.e. `false`.
As it doesn't seem supported in the code at present, can you move `econo`, `clock`, and `sleep` below the `// Not supported` line. That's just a minor style nitpick. Not important.